language_model/ngram_language_model.py

```diff
@@ -0,0 +1,168 @@
+        input=in_layer,
+        size=word_dim,
+        param_attr=paddle.attr.Param(
+            name="emb", initial_std=0.001, learning_rate=lr, l2_rate=0))
+    return word_embed
+
+
+def ngrm_network(voc_dim):
+    firWord = gen_embedding_layer("firWord", voc_dim)
+    secWord = gen_embedding_layer("secWord", voc_dim)
+    thirdWord = gen_embedding_layer("thirdWord", voc_dim)
+    forthWord = gen_embedding_layer("forthWord", voc_dim)
+    fifthWord = gen_embedding_layer("fifthWord", voc_dim)
+
+    fir = word_embed(firWord)
+    sec = word_embed(secWord)
+    third = word_embed(thirdWord)
+    forth = word_embed(forthWord)
+    fifth = word_embed(firWord)
+
+    embedding = paddle.layer.concat(input=[fir, sec, third, forth, fifth])
+
+    hidden1 = paddle.layer.fc(
+        input=embedding,
+        act=paddle.activation.Sigmoid(),
+        size=hidden_dim,
+        bias_attr=paddle.attr.Param(learning_rate=2 * lr),
+        layer_attr=paddle.attr.Extra(drop_rate=0.5),
+        param_attr=paddle.attr.Param(
+            learning_rate=lr, initial_std=1. / math.sqrt(word_dim * 8)))
+
+    output = paddle.layer.fc(
+        input=hidden1,
+        act=paddle.activation.Softmax(),
+        size=voc_dim,
+        bias_attr=paddle.attr.Param(learning_rate=2 * lr),
+        param_attr=paddle.attr.Param(learning_rate=lr))
+
+    return output
+
+
+def train_model(word_dict, voc_dim):
+    nxtWord = gen_embedding_layer("nxtWord", voc_dim)
+    output = ngrm_network(voc_dim)
+
+    def event_handler(event):
+        if isinstance(event, paddle.event.EndIteration):
+            if event.batch_id % 100 == 0:
+                result = trainer.test(
+                    paddle.batch(ptb.ngram_test(word_dict, 6), batch_size))
+                print("Pass %d, Batch %d, Cost %f, %s, Testing metrics %s" %
+                      (event.pass_id, event.batch_id, event.cost, event.metrics,
+                       result.metrics))
+        if isinstance(event, paddle.event.EndPass):
+            model_name = './models/model_pass_%05d.tar.gz' % event.pass_id
+            print("Save model into %s !" % model_name)
+            with gzip.open(model_name, 'w') as f:
+                parameters.to_tar(f)
+
+    cost = paddle.layer.classification_cost(input=output, label=nxtWord)
+    parameters = paddle.parameters.create(cost)
+
+    adadelta_optimizer = paddle.optimizer.AdaDelta(
+        learning_rate=0.1,
+        rho=0.95,
+        epsilon=1e-6,
+        model_average=paddle.optimizer.ModelAverage(
+            average_window=0.5, max_average_window=2500),
+        learning_rate_decay_a=0.0,
+        learning_rate_decay_b=0.0,
+        gradient_clipping_threshold=25)
+
+    trainer = paddle.trainer.SGD(cost, parameters, adadelta_optimizer)
+    trainer.train(
+        paddle.batch(ptb.ngram_train(word_dict, 6), batch_size),
+        num_passes=30,
+        event_handler=event_handler)
+
+
+def predict(word_dict, voc_dim):
```

Comment:
predict --> generate_sequence.

By using a language model to generate the sequence, it should be provided the first sentence. Here, how to get the input sentence?